sdk/cosmos/azure-cosmos-spark_3-1_2-12/pom.xml

```diff
@@ -55,7 +55,7 @@
     <dependency>
       <groupId>commons-io</groupId>
       <artifactId>commons-io</artifactId>
-      <version>2.4</version> <!-- {x-version-update;cosmos_commons-io:commons-io;external_dependency} -->
+      <version>2.7</version> <!-- {x-version-update;cosmos_commons-io:commons-io;external_dependency} -->
```

Comment:
Cosmos DB SDK relies on commons-io:commons-io and that's perfectly fine to upgrade that one.

However this one is not the SDK, but it is cosmos db spark connector, and as you see this is a provided dependency (next line). this being a provided dependency, cosmos db spark connector is not bringing in commons-io, but it will be provided by the spark platform.

You can see here: https://docs.databricks.com/release-notes/runtime/9.1.html that spark platform is still relying on commons-io 2.4 version.

we should revert this change for two reasons:
1. doesn't help anyone, as this is a provided dependency and at the end it is spark platform which is bringing in commons-io lib
2. this actually introduces a risk, as we may develop or test something on a library which is not the same as what is present on the spark platform.

Reply:
reverted this.